core/domain/user_services.py

```diff
@@ -931,24 +1021,175 @@ def create_new_user(gae_id, email):
+            modifiable_user_data object.
+    """
+
+    def _create_new_profile_transactional(
+            user_settings, user_auth_details):
+        """Save user models for new users as a transaction.
+
+        Args:
+            user_settings: UserSettings. The user settings domain object
+                corresponding to the newly created user.
+            user_auth_details: UserAuthDetails. The user auth details domain
+                object corresponding to the newly created list of users.
+        """
+        _save_user_auth_details(user_auth_details)
+        _save_user_settings(user_settings)
+
+    # As new profile user creation is done by a full (parent) user only.
+    parent_user_auth_details = get_auth_details_by_gae_id(gae_id, strict=True)
+    if parent_user_auth_details.pin is None:
+        raise Exception(
+            'Pin must be set for a full user before creating a profile.')
+    parent_user_id = parent_user_auth_details.user_id
+    user_settings_list = []
+    for modifiable_user_data in modifiable_user_data_list:
+        if modifiable_user_data.user_id:
+            raise Exception('User id cannot already exist for a new user.')
+        user_id = user_models.UserSettingsModel.get_new_id('')
+        user_settings = UserSettings(
+            user_id, gae_id, email, feconf.ROLE_ID_LEARNER,
+            preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE]
+        )
+        user_settings.populate_from_user_data(modifiable_user_data)
+
+        user_auth_details = UserAuthDetails(
+            user_id, None, modifiable_user_data.pin, parent_user_id)
+
+        transaction_services.run_in_transaction(
+            _create_new_profile_transactional, user_settings,
+            user_auth_details
+        )
+        user_settings_list.append(user_settings)
+    return user_settings_list
+
+
+def update_multiple_users_data(modifiable_user_data_list):
+    """Updates user settings and user auth model details for the users
+    specified in the modifiable_user_data_list.
+
+    Args: modifiable_user_data_list: list(ModifiableUserDataV1). The list of
+            modifiable_user_data entries corresponding to the users whose
+            data has to be updated.
+
+    Raises:
+        Exception. A valid user id does not exist for the user to be updated.
+        Exception. UserSettings or UserAuthDetail for a given user_id is
+            not found.
+    """
+    user_ids = [user.user_id for user in modifiable_user_data_list]
+    user_settings_list = get_users_settings(user_ids)
+    user_auth_details_list = get_multiple_user_auth_details(user_ids)
+    for (modifiable_user_data, user_settings, user_auth_details) in (
+            python_utils.ZIP(
+                modifiable_user_data_list, user_settings_list,
+                user_auth_details_list
+            )
+        ):
+        user_id = modifiable_user_data.user_id
+        if not user_id:
```

Comment:
if user_id is None

(The `''` should, conceptually, be handled in "User not found" below.)

Reply:
Yes correct, tested it and changed it.